### ISSUE TITLE
Terraform Import authentication error

### ISSUE THREAD

**Issue by Justin-DynamicD** [terraform-providers/terraform-provider-vsphere#1002]
### Terraform Version
v0.12.20

### vSphere Provider Version
Modules set to:
~> 1.12

actual running version:
1.16.2

### Affected Resource(s)
- `vsphere_virtual_machine`

Although, the error appears on login so I don't believe its module specific.

### Terraform Configuration/Scenario

Like I assume many do, my Terraform configuration contains a module which is called in order to deploy VMs and allow things like image names to change without forcing a destroy/redeploy of resources.

As such, the main configuration will call a module like so: 

```
module "sql" {
  source = "git@<git clone url>?ref=v1.6"

  VM-NamePrefix   = "qa-sql-"
  VM-Count = 1
  VM-CPU = 8
  VM-Memory = 32768
  VM-Template = "packer-win2016-latest"
}
```

Like-configured Vsphere modules can be found readily on registry.terraform.io

### Expected Behavior

Plan:
```
terraform plan
(...)
module.sql.vsphere_virtual_machine.vm[0]: Refreshing state... [id=xxx]
------------------------------------------------------------------------
------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.
```

Import:
```
terraform import module.sql.vsphere_virtual_machine.vm[1] xxx

(some kind of success message)
```

### Actual Behavior

Plan:
```
terraform plan
(...)
module.sql.vsphere_virtual_machine.vm[0]: Refreshing state... [id=xxx]
------------------------------------------------------------------------
------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.
```

Import:
```
terraform import module.sql.vsphere_virtual_machine.vm[1] xxx

Error: error setting up new vSphere SOAP client: ServerFaultCode: Cannot complete login due to an incorrect user name or password.
```

I'm not certain why I can plan just fine (including existing resource refresh) but no import. The login error seems correct in this case, but that's what it's returning so be it.

**Comment by bill-rich** [comment]
Hi @Justin-DynamicD!

That is odd that the plan is successful. It should be using the same client as on import. Would it be possible to get the debug log posted? See [here](https://www.terraform.io/docs/internals/debugging.html) for more info.

**Comment by Justin-DynamicD** [comment]
Thank you for the response.

After further testing I determined it was NOT an issue with the vSphere plugin, but instead just a behavior/limitation of Terraform 0.12.

The root cause: the root plan did not contain a provider definition, instead it was only defined in the module. BY updating the plan to define the provider in the root everything started working.

Basically Terraform 0.12 import doesn't play well with implicit providers.  I had to be explicit, pass those password in in the root of the playbook, and everything was happy.  I'm debating opening a ticket right to Terraform directly, but as that's a best practice anyway... I may chalk it up to a learning experience.
